src/UKHO.S100PermitService.API/Program.cs

```diff
@@ -166,7 +170,21 @@ private static void ConfigureServices(WebApplicationBuilder builder)
             builder.Services.AddSingleton<IHoldingsServiceAuthTokenProvider, AuthTokenProvider>();
             builder.Services.AddSingleton<IUserPermitServiceAuthTokenProvider, AuthTokenProvider>();
             builder.Services.AddSingleton<IProductKeyServiceAuthTokenProvider, AuthTokenProvider>();
+           
+            builder.Services.AddSingleton<IManufacturerKeyService>(sp =>
+            {
+                var cacheProvider = sp.GetRequiredService<ICacheProvider>();
+                var config = sp.GetRequiredService<IOptions<ManufacturerKeyConfiguration>>();
+                var secretClient = sp.GetRequiredService <ISecretClient> ();
+                return new ManufacturerKeyService(config, cacheProvider, secretClient);
+            });
+            builder.Services.AddSingleton<ISecretClient>(sp =>
+            {
+                var config = sp.GetRequiredService<IOptions<ManufacturerKeyConfiguration>>().Value;
+                return new KeyVaultSecretClient(new Uri(config.ServiceUri));
```

Comment:
Why are we explicitly passing ServiceUri from progarm.cs with class initilization. We can get value with IOptions<ManufacturerKeyConfiguration> from constructor parameter

Reply:
Changed he implementation. Configured n KeyVaultSecretClient.cs